presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -903,13 +996,26 @@ public static long truncateShort(
             return num - remainder;
         }
 
+        @LiteralParameters({"p", "s"})
+        @SqlType("decimal(p, s)")
+        public static long truncateShortBigintDecimals(
```

Comment:
This should be called `...BigintRoundScale`. Maybe rename argument?

Reply:
Have a look at 
```
@LiteralParameters({"p", "s", "rp"})
@SqlType("decimal(rp, s)")
@Constraint(variable = "rp", expression = "min(38, p + 1)")
public static Slice roundNLong(
        @LiteralParameter("s") long numScale,
        @LiteralParameter("rp") long resultPrecision,
        @SqlType("decimal(p, s)") Slice num,
        @SqlType(StandardTypes.BIGINT) long decimals)
```

if a rename `long decimals` → `scale`, then it might sound like the scale of the result (which it is not).

Since your comment about the method name applies only to the method that are intended to exist temporarily, let's go with the current name.